specification/notificationhubs/Notificationhubs.Management/models.tsp

```diff
@@ -0,0 +1,1600 @@
+  pnsCredentials?: PnsCredentials;
+
+  /**
+   * Gets or sets endpoint you can use to perform NotificationHub
+   * operations.
+   */
+  @visibility(Lifecycle.Read)
+  serviceBusEndpoint?: string;
+
+  /**
+   * Private Endpoint Connections for namespace
+   */
+  @visibility(Lifecycle.Read)
+  privateEndpointConnections?: PrivateEndpointConnectionResource[];
+
+  /**
+   * Gets or sets scaleUnit where the namespace gets created
+   */
+  @visibility(Lifecycle.Read, Lifecycle.Create)
+  scaleUnit?: string;
+
+  /**
+   * Deprecated.
+   */
+  dataCenter?: string;
+
+  /**
+   * Type of public network access.
+   */
+  publicNetworkAccess?: PublicNetworkAccess = PublicNetworkAccess.Enabled;
+}
+
+/**
+ * A collection of network authorization rules.
+ */
+model NetworkAcls {
+  /**
+   * List of IP rules.
+   */
+  @OpenAPI.extension("x-ms-identifiers", #[])
+  ipRules?: IpRule[];
+
+  /**
+   * A default (public Internet) network authorization rule, which contains rights if no other network rule matches.
+   */
+  publicNetworkRule?: PublicInternetAuthorizationRule;
+}
+
+/**
+ * A network authorization rule that filters traffic based on IP address.
+ */
+model IpRule {
+  /**
+   * IP mask.
+   */
+  @minLength(1)
+  ipMask: string;
+
+  /**
+   * List of access rights.
+   */
+  rights: AccessRights[];
+}
+
+/**
+ * A default (public Internet) network authorization rule, which contains rights if no other network rule matches.
+ */
+model PublicInternetAuthorizationRule {
+  /**
+   * List of access rights.
+   */
+  rights: AccessRights[];
+}
+
+/**
+ * Private Endpoint Connection properties.
+ */
+model PrivateEndpointConnectionProperties {
+  /**
+   * State of Private Endpoint Connection.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-provisioning-state" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  provisioningState?: PrivateEndpointConnectionProvisioningState;
+
+  /**
+   * Represents a Private Endpoint that is connected to Notification Hubs namespace using Private Endpoint Connection.
+   */
+  privateEndpoint?: RemotePrivateEndpointConnection;
+
+  /**
+   * List of group ids. For Notification Hubs, it always contains a single "namespace" element.
+   */
+  @visibility(Lifecycle.Read)
+  groupIds?: string[];
+
+  /**
+   * State of the Private Link Service connection.
+   */
+  privateLinkServiceConnectionState?: RemotePrivateLinkServiceConnectionState;
+}
+
+/**
+ * Represents a Private Endpoint that is connected to Notification Hubs namespace using Private Endpoint Connection.
+ */
+model RemotePrivateEndpointConnection {
+  /**
+   * ARM resource ID of the Private Endpoint. This may belong to different subscription and resource group than a Notification Hubs namespace.
+   */
+  @visibility(Lifecycle.Read)
+  id?: string;
```

Comment:
>string [](http://example.com/codeflow?start=7&length=6)

should be annoted as a resource id right, something like x-ms-resource-id as we have in swagger?

Reply:
`x-ms-resource-id` is a client behavior, shouldn't be part of service's spec.